app/src/main/kotlin/com/backbase/golden_sample_app/presentation/bottom/BottomNavigationDestinations.kt

```diff
@@ -0,0 +1,51 @@
+package com.backbase.golden_sample_app.presentation.bottom
+
+import androidx.annotation.IdRes
+import androidx.navigation.NavController
+import androidx.navigation.fragment.FragmentNavigator
+import androidx.navigation.fragment.FragmentNavigatorDestinationBuilder
+import com.backbase.android.design.header.TabHeaderFragment
+import com.backbase.android.design.header.TabListConfiguration
+import com.backbase.android.design.header.TopBarConfiguration
+import com.backbase.golden_sample_app.R
+import com.backbase.golden_sample_app.presentation.header.TabListConfigurationProvider
+import com.backbase.golden_sample_app.presentation.header.TopBarConfigurationProvider
+
+fun NavController.addBottomBarNavigationDestinations(
```

Comment:
Dashboard is added programatically while all the other screens are added via xml. This leaves with 2 different implementations without a clear explanation which direction we are going.

Reply:
My original idea was to add these 2 destinations in the XML graph. Unfortunately, due to some technical limitations on the `WorkspaceSelectorRouting`, I couldn't. 

`WorkspaceSelectorRouting` decides the navigation after the workspace is determined. They provide a hook function `onWorkspaceSelectedV2`, in which developers can decide where to navigate to based on the `workspaceInfo`. Unfortunately, this [function](https://github.com/Backbase/golden-sample-app-android/blob/main/app/src/main/kotlin/com/backbase/golden_sample_app/router/WorkspaceSelectorRoutingImpl.kt#L39) doesn't allow to specify a `Bundle`, so we cannot add the parameters needed by the `TabHeaderFragment` (this fragment receives a bundle with the top bar configuration and the tab list).

In order to solve this limitation on the `WorkspaceSelectorRouting`, I need to create the destination programmatically and provide them default values for the arguments.